### PR TITLE
Css preprocessor

### PR DESCRIPTION
**Pull request by danjee**
Rework by putting the less compiler as local variable for non thread safe access.

Add a fluent method for configuring minimized mode.

### CODE REVIEW

**Comment by danjee** [comment]
I'm not that pleased myself with the dumping of css result in a temporary file but I needed an URL for returning from getResourceUrl method. Is there a way to return a stream there where and put a content type on it ?

File: pippo-less4j/src/main/java/ro/pippo/less4j/LessResourceHandler.java [hunk 1]
```diff
@@ -39,18 +39,12 @@
...
     private static final Logger log = LoggerFactory.getLogger(LessResourceHandler.class);
 
-    private final LessCompiler compiler;
-
     private Map<String, String> sourceMap = new ConcurrentHashMap<>(); // cache
 
-    public LessResourceHandler(String urlPath, String resourceBasePath) {
-        this(urlPath, resourceBasePath, new ThreadUnsafeLessCompiler());
-    }
+    private boolean compress;
```
Comment (decebals) [inline pippo-less4j/src/main/java/ro/pippo/less4j/LessResourceHandler.java]: An idea is to initialize this variable in constructor
```
this.compress = Application.get().getPippoSettings().isProd()
```

**Comment by decebals** [comment]
The actual implementation of `StaticResourceHandler` works with URL. We try to find a solution, see https://github.com/decebals/pippo/pull/170#discussion_r33039265

**Comment by decebals** [comment]
A comment like "initialize compress" for the last commit would have been more appropriate.

**Comment by decebals** [comment]
This commit is a temporary solution. Thanks

**Pull request merged**